src/components/Counter/reducer.ts

```diff
@@ -1,32 +1,20 @@
-import * as R from 'ramda'
-import { combineReducers } from 'redux'
 import { createReducer } from 'redux-create-reducer'
 import { getType } from 'typesafe-actions'
 import * as actions from './actions'
-import { decrease, increase, save } from './helpers'
 
-export interface DayState {
-  pullUps: number
+export interface PullUpSession {
   totalPullUps: number
+  date: number
 }
+export type PullUpHistory = PullUpSession[]
 
-export const pullUpsReducer = createReducer(0, {
-  [getType(actions.increasePullUps)]: increase,
-  [getType(actions.decreasePullUps)]: decrease,
-  [getType(actions.resetPullUps)]: R.always(0),
+export const sessionReducer = createReducer<PullUpHistory, any>([], {
+  [getType(actions.saveSession)]: (state, action) => {
+    return [
+      ...state,
+      { totalPullUps: action.payload, date: createUnixTimeStamp() },
```

Comment:
Thats ugly, since it makes the reducer impure. Have to figure out an other way to pass the timestamp.